cranelift/filetests/filetests/isa/x64/b1.clif

```diff
@@ -99,9 +99,108 @@ block0(v0: i32):
 ;   pushq   %rbp
 ;   movq    %rsp, %rbp
 ; block0:
-;   shrl    $31, %edi, %edi
+;   sarl    $31, %edi, %edi
```

Comment:
Hmm, why it uses signed right shift now? It should be an unsigned shift

Reply:
This was with an eye towards better supporting our boolean representations, as mentioned in #3205, as `sar` will do sign extension.

Reply:
I see

Reply:
Given that it's not clear to me that we can use the wider boolean types with `icmp`, I've switched back to `shr` with a type guard on the result. Sorry about the back and forth here!